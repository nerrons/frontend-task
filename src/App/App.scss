@import '../responsive.scss';
@import '../index.scss';

.App {
  background-color: $primary-background;
  padding: 3%;

  @include sm {
    margin: 0 0;
  }
  @include md {
    margin: 0 10%;
  }
  @include lg {
    margin: 0 20%;
  }
  @include xl {
    margin: 0 25%;
  }
}
