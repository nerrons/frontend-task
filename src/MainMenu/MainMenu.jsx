import React from 'react';
import { Row, Col } from 'react-materialize';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './MainMenu.scss';

// A menu is a title plus a list of entries.
const MainMenu = ({ title, entries }) => (
  <div className="main-menu">
    <Title>{title}</Title>
    <hr />
    {entries.map(({ name, description }) => (
      <div className="entry-with-hr" key={name}>
        <Entry
          name={name}
          description={description}
        />
        <hr />
      </div>
    ))}
    <NavLink to="/">Back to main menu</NavLink>
  </div>
);
MainMenu.propTypes = {
  title: PropTypes.string.isRequired,
  entries: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
    }),
  ).isRequired,
};

const Title = ({ children }) => (
  <div className="main-title">
    <Row>
      <Col s={12}>{children}</Col>
    </Row>
  </div>
);
Title.propTypes = {
  children: PropTypes.string.isRequired,
};


// An entry consists of the name and the description.
const Entry = ({ name, description }) => (
  <div className="entry">
    <NavLink to={`${name.toLowerCase()}`}>
      <EntryName>{name}</EntryName>
      <EntryDescription>{description}</EntryDescription>
    </NavLink>
  </div>
);
Entry.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};


const EntryName = ({ children }) => (
  <div className="entry-name">
    <Row>
      <Col s={12} className="entry-name">{children}</Col>
    </Row>
  </div>
);
EntryName.propTypes = {
  children: PropTypes.string.isRequired,
};


const EntryDescription = ({ children }) => (
  <div className="entry-description">
    <Row>
      <Col s={12} className="entry-description">{children}</Col>
    </Row>
  </div>
);
EntryDescription.propTypes = {
  children: PropTypes.string.isRequired,
};


export default MainMenu;
