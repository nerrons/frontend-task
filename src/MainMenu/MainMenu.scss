.main-title {
  font-size: 3rem;
  color: #1a237e;
  line-height: 5rem;
}

.entry-name {
  font-size: 1.9rem;
  color: #1097d2;
  line-height: 4rem;
}

.entry-description {
  color: #1a237e;
  line-height: 2rem;
}