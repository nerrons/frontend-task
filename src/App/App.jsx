import React from 'react';
import { HashRouter, Route } from 'react-router-dom';
import MainMenu from '../MainMenu/MainMenu';
import SalesMenu from '../SalesMenu/SalesMenu';
import './App.scss';

const entries = [
  {
    name: 'Sales',
    description: 'Insights about sales',
  },
  {
    name: 'Customers',
    description: 'Insights about customers',
  },
  {
    name: 'Bulletin',
    description: 'Daily summary for your business',
  },
  {
    name: 'Ask',
    description: 'Ask a question',
  },
  {
    name: 'Tasks',
    description: 'Tools for your business',
  },
  {
    name: 'Connect',
    description: 'Connect...',
  },
];

const KonigleMenu = () => (
  <MainMenu title="Konigle" entries={entries} />
);

const App = () => (
  <HashRouter>
    <div className="App">
      <Route exact path="/" component={KonigleMenu} />
      <Route exact path="/sales" component={SalesMenu} />
    </div>
  </HashRouter>
);

export {
  entries,
};

export default App;
