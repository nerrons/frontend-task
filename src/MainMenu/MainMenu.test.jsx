import React from 'react';
import ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom'; // needed because MainMenu uses <NavLink>
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainMenu from './MainMenu';
import { entries } from '../App/App';

Enzyme.configure({ adapter: new Adapter() });

describe('MainMenu', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <HashRouter>
        <MainMenu
          title="Konigle"
          entries={entries}
        />
      </HashRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('has a valid snapshot', () => {
    const tree = renderer.create(
      <HashRouter>
        <MainMenu
          title="Konigle"
          entries={entries}
        />
      </HashRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders 6 entries', () => {
    const wrapper = shallow(
      <HashRouter>
        <MainMenu
          title="Konigle"
          entries={entries}
        />
      </HashRouter>,
    );
    const numEntries = wrapper.find(MainMenu).dive().find('.entry-with-hr').length;
    expect(numEntries).toBe(6);
  });
});
