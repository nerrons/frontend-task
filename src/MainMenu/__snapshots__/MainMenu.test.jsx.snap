// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu has a valid snapshot 1`] = `
<div
  className="main-menu"
>
  <div
    className="main-title"
  >
    <div
      className="row"
    >
      <div
        className="col s12"
      >
        Konigle
      </div>
    </div>
  </div>
  <hr />
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/sales"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Sales
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Insights about sales
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/customers"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Customers
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Insights about customers
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/bulletin"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Bulletin
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Daily summary for your business
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/ask"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Ask
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Ask a question
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/tasks"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Tasks
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Tools for your business
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
  <div
    className="entry-with-hr"
  >
    <div
      className="entry"
    >
      <a
        aria-current={null}
        href="#/connect"
        onClick={[Function]}
      >
        <div
          className="entry-name"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-name"
            >
              Connect
            </div>
          </div>
        </div>
        <div
          className="entry-description"
        >
          <div
            className="row"
          >
            <div
              className="col s12 entry-description"
            >
              Connect...
            </div>
          </div>
        </div>
      </a>
    </div>
    <hr />
  </div>
</div>
`;
